tools/online_editor/src/preview_widget.ts

```diff
@@ -44,6 +47,13 @@ export class PreviewWidget extends Widget {
 
     this.#canvas_id = "";
     this.#instance = null;
+    this.#resolve_attached_to_dom = () => {
+      // dummy, to be replaced with resolution function provided to promise
+      // executor.
+    };
+    this.#ensure_attached_to_dom = new Promise((resolve) => {
+      this.#resolve_attached_to_dom = resolve;
+    });
```

Comment:
This one took me a while! Clever!

Reply:
I wish it was simpler actually :-). But I couldn't find anything easier.